phoenix/app/cad_products_phoenix/lib/cad_products_phoenix_web/controllers/report_controller.ex

```diff
@@ -0,0 +1,32 @@
+defmodule CadProductsPhoenixWeb.ReportController do
+  use CadProductsPhoenixWeb, :controller
+
+  action_fallback CadProductsPhoenixWeb.FallbackController
+
+  alias CadProductsPhoenix.Management
+
+  def index(conn, _) do
+    products = product_json(Management.list_register())
+    path = "lib/cad_products_phoenix_web/reports/report_products.csv"
+
+    case Exq.enqueue(Exq, "report", CadProductsPhoenixWeb.Jobs.CreateReportJob, [products]) do
```

Comment:
Quando se trabalha com processamento em background uma das convenções é empilhar o job em background com o mínimo de informações possível, já que esse irá ocupar memória de cache enquanto não for processado.